tests/Unit/Core/ConstraintValidator/IsUrlRewriteValidatorTest.php

```diff
@@ -0,0 +1,170 @@
+
+/**
+ * Class IsUrlRewriteValidatorTest
+ */
+class IsUrlRewriteValidatorTest extends ConstraintValidatorTestCase
+{
+    /**
+     * @var bool
+     */
+    private $useAscendedChars;
+
+    public function setUp()
+    {
+        $this->useAscendedChars = false;
+
+        parent::setUp();
+    }
+
+    public function testItThrowsUnexpectedTypeExceptionOnIncorrectConstraintProvided()
+    {
+        $this->expectException(UnexpectedTypeException::class);
+
+        $this->validator->validate('valid-value', new NotBlank());
+    }
+
+    /**
+     * @dataProvider getIncorrectTypeRewriteUrls
+     */
+    public function testItThrowsUnexpectedTypeExceptionOnIncorrectValueTypeProvided($incorrectTypeRewriteUrl)
+    {
+        $this->expectException(UnexpectedTypeException::class);
+
+        $this->validator->validate($incorrectTypeRewriteUrl, new IsUrlRewrite());
+    }
+
+    /**
+     * @dataProvider getIncorrectRewriteUrls
+     */
+    public function testItFindsIncorrectUrlRewritePattern($incorrectRewriteUrl)
+    {
+        $this->validator->validate($incorrectRewriteUrl, new IsUrlRewrite());
+
+        $this->buildViolation((new IsUrlRewrite())->message)
+            ->setParameter('%s', '"' . $incorrectRewriteUrl . '"')
+            ->assertRaised()
+        ;
+    }
+
+    /**
+     * @dataProvider getCorrectRewriteUrls
+     */
+    public function testItFindsCorrectUrlRewritePatterns($correctRewriteUrl)
+    {
+        $this->validator->validate($correctRewriteUrl, new IsUrlRewrite());
+
+        $this->assertNoViolation();
+    }
+
+    /**
+     * @dataProvider getCorrectRewriteUlrUsingAscendingChars
+     */
+    public function testItFindsCorrectUrlRewritePatternUsingAscendedChars($correctRewriteUrl)
+    {
+        $this->useAscendedChars = true;
+
+        $validator = $this->createValidator();
+
+        $validator->validate($correctRewriteUrl, new IsUrlRewrite());
+
+        $this->assertNoViolation();
+    }
+
+    public function getIncorrectTypeRewriteUrls()
+    {
+        return [
+            [
+                [],
+            ],
+            [
+                true,
+            ],
+        ];
+    }
+
+    public function getIncorrectRewriteUrls()
+    {
+        return [
+            [
+                'test@!',
+            ],
+            [
+                '*test2*',
+            ],
+            [
+                'TęstĄČĘĖ',
+            ],
+            [
+                'tes/t/001',
+            ],
+        ];
+    }
+
+    public function getCorrectRewriteUrls()
+    {
+        return [
+            [
+                'my-test',
+            ],
+            [
+                'test',
+            ],
+            [
+                '123-589-test',
+            ],
+        ];
+    }
+
+    public function getCorrectRewriteUlrUsingAscendingChars()
+    {
+        return [
+            [
+              'aĮأ',
+            ],
+            [
+              'Šarūnas',
```

Comment:
😄

Reply:
I would have used mine but my name does not have `ascending chars` :cry: